src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -1448,6 +1475,10 @@ void Connection::receive_ident_message(session_ident_type session_ident, SaltedF
 {
     Session* sess = get_session(session_ident);
     if (REALM_UNLIKELY(!sess)) {
+        if (check_session_history(session_ident)) {
```

Comment:
do we need this? can't we check the state of the session?

Reply:
and if the session is not active anymore it's just a no-op

Reply:
nvm, I see we don't have a session in this case..

Reply:
It really isn't a client error if there is a pending network message that is received after the session has been deactivated, so there is no need to close the connection and restart any other sessions that are currently active. In this case we just print message (maybe it should be an info and not an error?) and just ignore it.
This is really only needed if session multiplexing is enabled, and doesn't matter if it is disabled, since there will only ever be one session per connection.

Reply:
Got it. I have a suggestion (feel free to ignore). We could change `close_due_to_*` (actually disconnect()) to use `m_sessions` (or similar) to decide if it's the last session wanting to disconnect, and only then actually close the connection (a bit similar to the logic in `one_more_active_unsuspended_session`). We then don't need the new check (and error message), although I agree the new error message makes more sense.

Reply:
I agree - but I think it would be best in a separate PR.